srcs/class/response.cpp

```diff
@@ -6,148 +6,157 @@
-		this->set_status( 400, "Bad Request" );
-	else if ( this->version != req.get_http_version() )
-		this->set_status( 505, "HTTP Version Not Supported" );
+	if (req.is_request_valid() == false)
+		this->set_status(400, "Bad Request");
+	else if (this->version != req.get_http_version())
+		this->set_status(505, "HTTP Version Not Supported");
 }
 
-Response::~Response( void ) {}
+Response::~Response(void)
+{
+}
 
-std::string Response::get_str_code(void) {
+std::string Response::get_str_code(void)
+{
 
-	return to_string( this->status_code );
+	return to_string(this->status_code);
 }
 
-int	Response::add_header( std::string key, std::string value ) {
+int Response::add_header(std::string key, std::string value)
+{
 
-	this->headers.insert( headers_t::value_type(key, value) );
+	this->headers.insert(headers_t::value_type(key, value));
 	return (1);
 }
 
-
-void	Response::set_status( int status_code, std::string msg ) {
+void Response::set_status(int status_code, std::string msg)
+{
 	this->status_code = status_code;
 	this->status_message = msg;
 }
 
-int	Response::send() {
+int Response::send()
+{
 
 	/* add some headers */
-	http_header_content_length( this->req, *this );
+	http_header_content_length(this->req, *this);
 
 	std::string raw_response;
 
-	raw_response = this->version + " " + to_string( this->status_code ) + " " + this->status_message;
-	for ( headers_t::iterator it = this->headers.begin(); it != this->headers.end(); ++it )
+	raw_response = this->version + " " + to_string(this->status_code) + " " + this->status_message;
+	for (headers_t::iterator it = this->headers.begin(); it != this->headers.end(); ++it)
 		raw_response += "\n" + it->first + ": " + it->second;
 	raw_response += "\r\n\n";
 	raw_response += body;
 
-
-	#ifdef PRINT_REQ
-		std::cout << "<====>" << std::endl;
-		std::cout << raw_response << std::endl;
-		std::cout << "<====>" << std::endl;
-	#endif
+#ifdef PRINT_REQ
+	std::cout << "<====>" << std::endl;
+	std::cout << raw_response << std::endl;
+	std::cout << "<====>" << std::endl;
+#endif
 
 	int status = ::send(this->client_socket, raw_response.c_str(), raw_response.size(), 0);
 
-
 	return (status);
 }
 
-std::string auto_index_template( std::string url, std::string legacy_url );
+std::string auto_index_template(std::string url, std::string legacy_url);
 
-std::string Response::load_body( Request &req ) {
+std::string Response::load_body(Request &req)
+{
 
 	std::string new_body;
-	if ( req.auto_index == true ) {
+	if (req.auto_index == true)
+	{
 		this->add_header("Content-Type", "text/html");
-		new_body = auto_index_template( req.getUrl(), req.get_legacy_url() );
+		new_body = auto_index_template(req.getUrl(), req.get_legacy_url());
 	}
-	else if ( req.get_route().cgi_enable == true 
-		&& get_extension( req.getUrl().c_str() ) == req.get_route().cgi_extension ) {
+	else if (req.get_route().get_cgi_enable() == true && get_extension(req.getUrl().c_str()) == req.get_route().get_cgi_extension())
```

Comment:
Tu pourrai essayer ça comme ça? Les arguments peuvent être déduis par le compilateur ;)

```suggestion
	else if (req.get_route().get_cgi_enable() && get_extension(req.getUrl()) == req.get_route().get_cgi_extension())
```